htdocs/api/index.php

```diff
@@ -172,6 +172,22 @@
 // Restrict API to some IPs
 if (!empty($conf->global->API_RESTRICT_ON_IP)) {
 	$allowedip = explode(' ', $conf->global->API_RESTRICT_ON_IP);
+
+	// allow the use of FQDN for dynamic dns access
+	$count = count($allowedip);
+	for ($i = 0; $i < $count; $i++) {
+		// if submitted host is not an IP but some FQDN
+		if (!preg_match('/^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/', $allowedip[$i])
+			&& preg_match('/^[a-zA-Z0-9-]*\.[a-zA-Z0-9-]*\.[a-zA-Z0-9-]*\.$/', $allowedip[$i])) {
+			// set the IP of the host
+			$allowedip[$i] = gethostbyname($allowedip[$i]);
+			//  if submitted host is not an IP removes it
+		} elseif (!preg_match('/^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/', $allowedip[$i])) {
```

Comment:
i think we should remove this elseif.
Ip entered may also be an ipv6.
And cleaning the array $allowedip[$i] seems useless because value is compared later to ipv4 and ipv6 so if value is wrong, it will just not match any test later.

Reply:
It's OK for me. I'll do it in next days.
Thanks

Reply:
@eldk Time to make the suggested minor fix so i can validate the PR ?